apache/superset
# chore: refactor header menu to show in header grid component

### SUMMARY
This pr removes the hovermenu less file and refactors code to show the dashboard hover menu delete icon inside the header. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

before

https://user-images.githubusercontent.com/17326228/133115922-7670a640-18f5-4649-abc5-38920d63a3c0.mov

after

https://user-images.githubusercontent.com/17326228/133116020-c83fc7a5-20c9-4a6f-a79c-c22472f688b5.mov


### TESTING INSTRUCTIONS
Go to a dashboard and click on edit. Next drag a header tab to dashboard and ensure that the delete button show up on hover.

### ADDITIONAL INFORMATION


- [x ] Has associated issue: Fixes https://github.com/apache/superset/issues/16424
